packages/react-core/src/modules/guide/context/KnockGuideProvider.tsx

```diff
@@ -0,0 +1,60 @@
+import Knock, {
+  KnockGuideClient,
+  KnockGuideTriggerParams,
+} from "@knocklabs/client";
+import * as React from "react";
+
+import { useKnockClient, useStableOptions } from "../../core";
+import { ColorMode } from "../../core/constants";
+
+type KnockGuideProviderValue = {
+  client: KnockGuideClient;
+  colorMode: ColorMode;
+};
+
+export const KnockGuideContext = React.createContext<
+  KnockGuideProviderValue | undefined
+>(undefined);
+
+type Props = KnockGuideTriggerParams & {
+  colorMode?: ColorMode;
+};
+
+export const KnockGuideProvider: React.FC<React.PropsWithChildren<Props>> = ({
+  colorMode = "light",
+  data,
+  tenant,
+  children,
+}) => {
+  let knock: Knock;
+
+  try {
+    knock = useKnockClient();
+  } catch (_) {
+    throw new Error("KnockGuideProvider must be used within a KnockProvider");
+  }
+
+  const triggerParams = useStableOptions({ data, tenant });
+
+  const knockGuideClient = React.useMemo(() => {
+    return new KnockGuideClient(knock, triggerParams);
```

Comment:
I think this bit here is worth flagging and drawing attention for feedbacks: trigger params are a part of the guide client constructor arg, which are passed in as props to the provider and used in a dep array for memoizing the client. 

This means, if trigger params change (e.g. `data` or `tenant`) then we re-instantiate the client effectively resetting it. This is different than filter params like `message_type` that can be passed in to the fetch function in an ad hoc fashion, because trigger params affect eligibility evaluations as well as the rendered content. This way we make sure each instantiated client can have guides that are consistently evaluated and rendered.